ERCS/erc-7828.md

```diff
@@ -0,0 +1,236 @@
+
+In the same spirit, the address could be a human-readable name as well, which is already a use case for ENS. However it would be desirable if the address name could be registered on a L2.
+
+Desired properties:
+- a unified format to represent any address on L1 or L2
+- the ability to use chain names in addition to identifiers
+- the chain portion can be a domain name, or just the suffix for a "base chain" (eg. `eth`, `myfavoriterollup.eth`, `sepolia`, `my_l3.base.superchain.eth`)
+- the address portion can be either the appropriate type of address for the chain (0x... for EVM chains, otherwise eg. for starknet something else), or a domain name (ENS or other)
+- the ability to resolve ENS names on the L2
+- the address portion and the chain portion should be resolved separately 
+- checksums are MANDATORY
+- checksum hash goes over the entire address, so users can't just replace a component and expect it to stay valid
+
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Format
+
+Valid addresses MUST include the identifier of the chain they belong to.
+
+```
+L1-TLD ::= eth | sepolia | …
+chain_id ::= 0x[a-fA-F0-9]{1,64}
+address ::= 0x[a-fA-F0-9]{40}
+chain ::= <chain_id> | <L1-TLD> | <ens-name> . <L1-TLD>
+user ::= <address> | <ens-subdomain>
+account ::= <user>@<chain>
+```
+
+Note the difference between `ens-name`, which is a full ENS name, and `ens-subdomain` that is just a segment of a name between dots. E.g. `user.app.eth` is a name, `user` and `app` are subdomains.
+
+A few examples below. 
+
+Option 1: using @ to separate address and chain
+```
+Mainnet
+- 0x12345...6789@0x1
+- 0x12345...6789@eth
+- alice.eth@eth
+
+Testnet (Sepolia)
+- 0x12345...6789@0xaa36a7
+- 0x12345...6789@sepolia
+- alice.eth@sepolia
+
+Rollup
+- 0x12345...6789@chainId
+- 0x12345...6789@arbitrum.eth
+- alice.eth@arbitrum.eth
+
+Rollup Sepolia
+- 0x12345...6789@arbitrum.sepolia
+
+My ENS name is registered on rollup1, but I want to receive funds on rollup2
+- alice.rollup1.eth@rollup2.eth
+```
+
+Option 2: using : instead of @ 
+```
+Mainnet
+- 0x12345...6789:0x1
+- 0x12345...6789:eth
+- alice.eth:eth
+
+Testnet (Sepolia)
+- 0x12345...6789:0xaa36a7
+- 0x12345...6789:sepolia
+- alice.eth:sepolia
+
+Rollup
+- 0x12345...6789:chainId
+- 0x12345...6789:arbitrum.eth
+- alice.eth:arbitrum.eth
+
+Rollup Sepolia
+- 0x12345...6789:arbitrum.sepolia
+
+My ENS name is registered on rollup1, but I want to receive funds on rollup2
+- alice.rollup1.eth:rollup2.eth
+```
+### CHECKSUM
+<!-- TODO: add more explanation here -->
+
+Two desired properties:
+1) checksums are MANDATORY
+2) checksum hash goes over the entire address, so users can't just go and replace a component and expect it to stay valid
```

Comment:
This type of content belongs in the Rationale section, as it is describing why you made particular choices. Keep the Specification section as close to just requirements as possible so it's easy for an implementer to follow.